lib/optimization/table_data.go

```diff
@@ -131,17 +131,27 @@ func (t *TableData) InsertRow(pk string, rowData map[string]any, delete bool) {
 	prevRow, isOk := t.rowsData[pk]
 	if isOk {
 		prevRowSize = size.GetApproxSize(prevRow)
-		for key, val := range rowData {
-			if val == constants.ToastUnavailableValuePlaceholder {
-				// Copy it from prevRow.
-				prevVal, isOk := prevRow[key]
-				if !isOk {
-					continue
+		if delete {
+			// If the row was deleted, preserve the previous values if we have them in memory
+			rowData = prevRow
+			if len(rowData) == 0 {
```

Comment:
How would we have a len(rowData) == 0 ?

Reply:
I'm pretty sure the only way this would happen is if we had a [tombstone message](https://docs.google.com/document/d/1ApCs6EewoIlUUHkJGDAJLOs7f13AyB2AhuAAk020FJM/edit), but we're skipping those now

Reply:
A [test was panicking](https://github.com/artie-labs/transfer/actions/runs/9670133429/job/26678253059#step:5:53) because it was inserting a row that was a nil map. Initially I made the test provide an empty map instead of nil, but then I thought it would be safer to handle that here so we won't get a panic if a nil row does get inserted for any reason.

Reply:
Hm got it. Do you want to throw an error instead?

Reply:
Or get rid of the test?